eppo_core/src/event_ingestion/event_delivery.rs

```diff
@@ -78,13 +126,100 @@ impl EventDelivery {
             }
         })?;
         let response = response
-            .json::<EventDeliveryResponse>()
+            .json::<RawEventDeliveryResponse>()
             .await
-            .map_err(EventDeliveryError::NonRetriableError)?;
+            .map_err(|e| JsonDeserializationError(e))?;
         info!(
             "Batch delivered successfully, {} events failed ingestion",
             response.failed_events.len()
         );
-        Ok(response)
+        Ok(EventDeliveryResponse {
+            retriable_failed_events: response.failed_events,
+            non_retriable_failed_events: failed_validation_events,
+        })
+    }
+}
+
+/// Returns whether the provided event's serialized JSON string is over the length limit
+fn ensure_max_event_size(event: &Event) -> Result<bool, EventDeliveryError> {
+    let serialized_event = serde_json::to_vec(event).map_err(|e| JsonSerializationError(e))?;
+    Ok(serialized_event.len() < MAX_EVENT_SERIALIZED_LENGTH)
+}
```

Comment:
minor: this is quite wasteful as it serializes the event but discards the work, so we have to go through serialization again